From 81b76056229c126eb1884d305be8a5ce23ad6ba1 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 11 Aug 2026 08:49:48 +0000
Subject: [PATCH] Re-order services so that it works even without qtaguid

Change-Id: I0c0f527b3ae151d45c68f7ac6c205da3f34e74df

[Port to A16] Same reorder; context refreshed for A16-added aconfig flag lines after the LocalServices block.

Change-Id: I945e8c17c09695a47662bb3c7d8d7b6f162dcff9
---
 .../android/server/net/NetworkPolicyManagerService.java   | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
index d81103ca52d5..aed7f4ca8388 100644
--- a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
+++ b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
@@ -1071,6 +1071,10 @@ public class NetworkPolicyManagerService extends INetworkPolicyManager.Stub {
         Trace.traceBegin(Trace.TRACE_TAG_NETWORK, "systemReady");
         final int oldPriority = Process.getThreadPriority(Process.myTid());
         try {
+            mUsageStats = LocalServices.getService(UsageStatsManagerInternal.class);
+            mAppStandby = LocalServices.getService(AppStandbyInternal.class);
+            mActivityManagerInternal = LocalServices.getService(ActivityManagerInternal.class);
+
             // Boost thread's priority during system server init
             Process.setThreadPriority(Process.THREAD_PRIORITY_FOREGROUND);
             if (!isBandwidthControlEnabled()) {
@@ -1078,10 +1082,6 @@ public class NetworkPolicyManagerService extends INetworkPolicyManager.Stub {
                 return;
             }
 
-            mUsageStats = LocalServices.getService(UsageStatsManagerInternal.class);
-            mAppStandby = LocalServices.getService(AppStandbyInternal.class);
-            mActivityManagerInternal = LocalServices.getService(ActivityManagerInternal.class);
-
             mUseMeteredFirewallChains = Flags.useMeteredFirewallChains();
             mUseDifferentDelaysForBackgroundChain = Flags.useDifferentDelaysForBackgroundChain();
             mNeverApplyRulesToCoreUids = Flags.neverApplyRulesToCoreUids();
-- 
2.43.0

